test/integration/ingress_regex_match_test.go

```diff
@@ -170,10 +170,21 @@ func TestIngressRegexMatchPath(t *testing.T) {
 
 			t.Log("testing paths expected not to match")
 			for _, path := range tc.notMatchPaths {
-				resp, err := helpers.DefaultHTTPClient().Get(fmt.Sprintf("%s%s", proxyURL, path))
-				require.NoError(t, err)
-				defer resp.Body.Close()
-				require.Equalf(t, http.StatusNotFound, resp.StatusCode, "should not match path %s: %s", path, tc.description)
+				require.Eventually(t, func() bool {
```

Comment:
Isn't that basically: https://github.com/Kong/kubernetes-ingress-controller/blob/3ce7c14afd2099e991c4dcd3370f23ef8c742a5e/test/internal/helpers/http.go#L51 ? 🤔

Reply:
That's a very good point. I followed the rabbit hole and replaced such calls at some other places too 👍